diff --git a/lxqt-config-monitor/kscreenutils.cpp b/lxqt-config-monitor/kscreenutils.cpp
index 9515e78..be2634d 100644
--- a/lxqt-config-monitor/kscreenutils.cpp
+++ b/lxqt-config-monitor/kscreenutils.cpp
@@ -2,6 +2,7 @@
 #include "timeoutdialog.h"
 
 #include <KScreen/Output>
+#include <KScreen/Mode>
 #include <KScreen/Config>
 #include <KScreen/GetConfigOperation>
 #include <KScreen/SetConfigOperation>
diff --git a/lxqt-config-monitor/loadsettings.cpp b/lxqt-config-monitor/loadsettings.cpp
index 0c7bd73..4e9331b 100644
--- a/lxqt-config-monitor/loadsettings.cpp
+++ b/lxqt-config-monitor/loadsettings.cpp
@@ -23,6 +23,7 @@
 #include "kscreenutils.h"
 #include <KScreen/Output>
 #include <KScreen/Config>
+#include <KScreen/Mode>
 #include <KScreen/GetConfigOperation>
 #include <KScreen/SetConfigOperation>
 #include <LXQt/Settings>
diff --git a/lxqt-config-monitor/monitorpicture.cpp b/lxqt-config-monitor/monitorpicture.cpp
index 0d06ab8..4cb1489 100644
--- a/lxqt-config-monitor/monitorpicture.cpp
+++ b/lxqt-config-monitor/monitorpicture.cpp
@@ -24,6 +24,7 @@
 #include <QDebug>
 #include <QVector2D>
 #include <QRectF>
+#include <KScreen/Mode>
 #include <QScrollBar>
 
 #include "configure.h"
diff --git a/lxqt-config-monitor/monitorsettingsdialog.cpp b/lxqt-config-monitor/monitorsettingsdialog.cpp
index 6172019..bfd8c1d 100644
--- a/lxqt-config-monitor/monitorsettingsdialog.cpp
+++ b/lxqt-config-monitor/monitorsettingsdialog.cpp
@@ -28,6 +28,7 @@
 #include "kscreenutils.h"
 
 #include <KScreen/Output>
+#include <KScreen/Mode>
 #include <QJsonObject>
 #include <QJsonArray>
 #include <LXQt/Settings>
diff --git a/lxqt-config-monitor/monitorwidget.cpp b/lxqt-config-monitor/monitorwidget.cpp
index e0fcf0a..41883c2 100644
--- a/lxqt-config-monitor/monitorwidget.cpp
+++ b/lxqt-config-monitor/monitorwidget.cpp
@@ -22,6 +22,7 @@
 #include <QComboBox>
 #include <QStringBuilder>
 #include <QDialogButtonBox>
+#include <KScreen/Mode>
 #include <KScreen/EDID>
 
 #include <algorithm>
