diff --git a/kernel/nv-drm.c b/usr/src/nvidia-drivers/nv-drm.c
index ba9fa61..72b3801 100644
--- a/kernel/nv-drm.c
+++ b/usr/src/nvidia-drivers/nv-drm.c
@@ -71,7 +71,11 @@ static const struct file_operations nv_drm_fops = {
 };
 
 static struct drm_driver nv_drm_driver = {
-    .driver_features = 0,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+    .driver_features = DRIVER_GEM | DRIVER_PRIME | DRIVER_LEGACY,
+#else
+     .driver_features = DRIVER_GEM | DRIVER_PRIME,
+#endif
     .load = nv_drm_load,
     .unload = nv_drm_unload,
     .fops = &nv_drm_fops,
diff --git a/kernel/nv-linux.h b/usr/src/nvidia-drivers/nv-linux.h
index 4482ac6..1b2f59e 100644
--- a/kernel/nv-linux.h
+++ b/usr/src/nvidia-drivers/nv-linux.h
@@ -1889,8 +1889,13 @@ static inline NvU64 nv_node_end_pfn(int nid)
         }
     #endif
 #else
-    #define NV_GET_USER_PAGES(start, nr_pages, write, force, pages, vmas) \
-        get_user_pages(current, current->mm, start, nr_pages, write, force, pages, vmas)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+     #define NV_GET_USER_PAGES(start, nr_pages, gup_flags, pages, vmas) \
+         get_user_pages(current, current->mm, start, nr_pages, gup_flags, pages, vmas)
+#else
+     #define NV_GET_USER_PAGES(start, nr_pages, write, force, pages, vmas) \
+         get_user_pages(current, current->mm, start, nr_pages, write, force, pages, vmas)
+#endif
 
     #define NV_GET_USER_PAGES_REMOTE    get_user_pages
 #endif
diff --git a/kernel/nv-mlock.c b/usr/src/nvidia-drivers/nv-mlock.c
index 0f795a5..9e3c5ea 100644
--- a/kernel/nv-mlock.c
+++ b/usr/src/nvidia-drivers/nv-mlock.c
@@ -50,7 +50,11 @@ RM_STATUS NV_API_CALL nv_lock_user_pages(
 
     down_read(&mm->mmap_sem);
     ret = NV_GET_USER_PAGES((unsigned long)address,
-                            page_count, write, force, user_pages, NULL);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+                             page_count, write ? FOLL_WRITE : 0, force, user_pages, NULL);
+#else
+                             page_count, write, force, user_pages, NULL);
+#endif
     up_read(&mm->mmap_sem);
     pinned = ret;
 
