diff --git a/CMakeLists.txt b/CMakeLists.txt.orig
index 684e34f..367740e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt.orig
@@ -100,7 +100,7 @@ if(BUILD_GAME OR BUILD_EDITOR)
 
 	find_package(Bullet REQUIRED QUIET)
 	include_directories(${BULLET_INCLUDE_DIRS})
-	list(APPEND LIBS ${BULLET_LIBRARIES})
+	list(APPEND LIBS ${BULLET_LIBRARIES} BulletFileLoader LinearMath)
 
 	find_package(SDL2 REQUIRED)
 	include_directories(${SDL2_INCLUDE_DIR})
