diff --git a/./code/sys/sys_local.h b/./code/sys/sys_local.h.new
index bfe270f..4ffefca 100644
--- a/./code/sys/sys_local.h
+++ b/./code/sys/sys_local.h.new
@@ -22,6 +22,7 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
 #include "../qcommon/q_shared.h"
 #include "../qcommon/qcommon.h"
+#include <unistd.h>
 
 // Require a minimum version of SDL
 #define MINSDL_MAJOR 1
